frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java

```diff
@@ -71,6 +71,8 @@ public void after(java.util.List<? extends VFileEvent> events) {
     setContent(ScrollPaneFactory.createScrollPane(graphTable));
 
     // The following listener executes on each opening of the Git Machete tab.
+    // Since we don't want to constantly ask the user about rediscovery the listener is removed after first execution.
+    // Final effect: execution only when user opens Git Machete tab for the first time during an IntelliJ session.
```

Comment:
Hmmm one more concern... what if there are multiple git repositories within the project (see lines 79-80)? won't the rediscovery just trigger for one of the git repositories &mdash; the one that happens to be selected when the Git Machete tab is first opened? or even for **no** repository at all, if none is selected (I don't remember now if it's realistically possible, nvm)

CC @mkondratek who implemented most of multi-git-repository handling

Reply:
After changing the order of checking operations to `Modification Date -> Are local branches managed` I also went back to the previous way reacting to the every opening of Git Machete tab. Now We can state that a suggestion has already been made or is not applicable by changing the modification date of the `.git/machete` file to `System.currentTimeMillis()`. The listener will react everytime so this should cover the multi-git-repo scenario now.